testing/xfpga/test_mock_errinj_c.cpp

```diff
@@ -0,0 +1,303 @@
+// Copyright(c) 2017-2018, Intel Corporation
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <opae/enum.h>
+#include <opae/properties.h>
+#include "intel-fpga.h"
+#include "gtest/gtest.h"
+#include "types_int.h"
+#include "test_system.h"
+#include <opae/mmio.h>
+#include <cstdarg>
+#include <linux/ioctl.h>
+#include "xfpga.h"
+
+#undef FPGA_MSG
+#define FPGA_MSG(fmt, ...) \
+	printf("MOCK " fmt "\n", ## __VA_ARGS__)
+
+using namespace opae::testing;
+
+
+int port_release_ioctl(mock_object * m, int request, va_list argp){
+  int retval = -1;
+  errno = EINVAL;
+  struct fpga_fme_port_release *port_release =
+         va_arg(argp, struct fpga_fme_port_release *);
+  if (!port_release) {
+    FPGA_MSG("port_release is NULL");
+    goto out_EINVAL;
+  }
+  if (port_release->argsz != sizeof(*port_release)) {
+    FPGA_MSG("wrong structure size");
+    goto out_EINVAL;
+  }
+  if (port_release->flags != 0) {
+    FPGA_MSG("unexpected flags %u", port_release->flags);
+    goto out_EINVAL;
+  }
+  if (port_release->port_id != 0) {
+    FPGA_MSG("unexpected port ID %u", port_release->port_id);
+    goto out_EINVAL;
+  }
+  retval = 0;
+  errno = 0;
+
+out:
+  va_end(argp);
+  return retval;
+
+out_EINVAL:
+  retval = -1;
+  errno = EINVAL;
+  goto out;
+}
+
+int port_assign_ioctl(mock_object * m, int request, va_list argp){
+  int retval = -1;
+  errno = EINVAL;
+  struct fpga_fme_port_assign *port_assign =
```

Comment:
Please add the following to appease g++ >= 4.8.5
UNUSED_PARAM(m);
UNUSED_PARAM(request);

Any way you could update your g++?